tracer/src/Datadog.Trace.Tools.Runner/RunCiCommand.cs

```diff
@@ -103,7 +105,8 @@ public override async Task<int> ExecuteAsync(CommandContext context, RunCiSettin
                 var lazyItrClient = new Lazy<IntelligentTestRunnerClient>(() => new(CIEnvironmentValues.Instance.WorkspacePath, ciVisibilitySettings));
                 if (ciVisibilitySettings.GitUploadEnabled != false || ciVisibilitySettings.IntelligentTestRunnerEnabled)
                 {
-                    await lazyItrClient.Value.UploadRepositoryChangesAsync().ConfigureAwait(false);
+                    // If we are in git upload only then we can defer the await until the child command exits.
+                    uploadRepositoryChangesTask = Task.Run(() => lazyItrClient.Value.UploadRepositoryChangesAsync());
```

Comment:
Seems if `Program.CallbackForTests` is null, the program will exit before this is run. Don't know enough to tell whether this is an issue or not?

Reply:
good one, I change it in [8195beb](https://github.com/DataDog/dd-trace-dotnet/pull/3825/commits/8195beb8f8b233a0d7b2bfc6081b0aecd47231bf)